src/v/features/enterprise_features.h

```diff
@@ -75,4 +78,60 @@ class enterprise_feature_report {
     vtype _disabled;
 };
 
+template<config::detail::Property P>
+class sanctioning_binding {
+public:
+    using T = P::value_type;
+    explicit sanctioning_binding(config::enterprise<P>& prop)
+      : _prop(prop)
+      , _binding(_prop.bind()) {
+        update_sanctioned_state();
+        _binding.watch([this] { update_sanctioned_state(); });
+    }
+
+    config::binding<T>& binding() { return _binding; }
+    const config::binding<T>& binding() const { return _binding; }
+
+    std::pair<T, bool> operator()(bool should_sanction) const {
+        const auto& val = _binding();
+        if (should_sanction && _is_sanctioned) [[unlikely]] {
+            return std::make_pair(_prop.default_value(), true);
+        } else {
+            return std::make_pair(val, false);
+        }
+    }
+
+private:
+    config::enterprise<P>& _prop;
+    config::binding<T> _binding;
+    bool _is_sanctioned{false};
+
+    void update_sanctioned_state() {
+        _is_sanctioned = _prop.check_restricted(_binding());
+    }
+};
+
+namespace details {
+template<typename T>
+concept NonVoid = !std::is_void_v<T>;
+
+} // namespace details
+template<license_required_feature F>
+details::NonVoid auto make_sanctioning_binding() {
+    if constexpr (
+      F == license_required_feature::partition_auto_balancing_continuous) {
+        return sanctioning_binding(
+          config::shard_local_cfg().partition_autobalancing_mode);
+    }
+
+    if constexpr (F == license_required_feature::core_balancing_continuous) {
+        return sanctioning_binding(
+          config::shard_local_cfg().core_balancing_continuous);
+    }
+
+    if constexpr (F == license_required_feature::leadership_pinning) {
+        return sanctioning_binding(
+          config::shard_local_cfg().default_leaders_preference);
+    }
+}
```

Comment:
Is this equivalent to just inlining the creation of the bindings? That seems simpler to me, so I would prefer that.

Reply:
While it is indeed an extra step, this explicitly ties `license_required_feature` to the enterprise properties.
I am not fully convinced that this is the best way to express this relationship, but i would prefer to get the behavior out of the way and revisit it once we don't have a tight deadline.

Preferably, i would like a direct relation between the `license_required_feature` and the enterprise property that represents that behavior. Note that `license_required_feature` extends beyond `config::properties`. The behavior is also reflected on topic configs and node configs. So, more thought is required.